sfm_pc/management/commands/import_google_doc.py

```diff
@@ -1744,24 +1744,45 @@ def create_person(self, person_data):
                 else:
                     if rank_name:
-                        rank, _ = Rank.objects.get_or_create(value=rank_name)
+                        rank, _ = Rank.objects.get_or_create(value=rank_name.strip())
                         rank = rank.id
                     else:
                         rank = None
 
                 try:
-                    title = person_data[membership_positions['Title']['value']] or None
+                    title = person_data[membership_positions['Title']['value']].strip() or None
                 except IndexError:
                     title = None
 
+                membership_kwargs = {
+                    'membershippersonmember__value': person,
+                    'membershippersonorganization__value': organization,
+                    'membershippersonfirstciteddate__value': fcd,
+                    'membershippersonlastciteddate__value': lcd,
+                    'membershippersonrole__value': role,
+                    'membershippersonrank__value': rank,
+                    'membershippersontitle__value': title,
+                }
+
                 try:
-                    membership = MembershipPerson.objects.get(membershippersonmember__value=person,
-                                                              membershippersonorganization__value=organization,
-                                                              membershippersonfirstciteddate__value=fcd,
-                                                              membershippersonlastciteddate__value=lcd,
-                                                              membershippersonrole__value=role,
-                                                              membershippersonrank__value=rank,
-                                                              membershippersontitle__value=title)
+                    membership = MembershipPerson.objects.get(**membership_kwargs)
+
+                    sources = set(self.sourcesList(membership, 'member') + self.sourcesList(membership, 'organization'))
+                    membership_data['MembershipPerson_MembershipPersonMember']['sources'] += sources
+                    membership.update(membership_data)
+
+                except MembershipPerson.MultipleObjectsReturned:
+                    memberships = MembershipPerson.objects.filter(**membership_kwargs)
+                    membership = memberships.order_by('id').first()
+                    duplicates = memberships.exclude(id=membership.id)
+
+                    self.stdout.write('Found {0} duplicate memberships for {1}'.format(duplicates.count(), person))
+                    remove_duplicates = input('Remove duplicates? [y/n] ')
+
+                    if remove_duplicates.lower().strip() == 'y':
+                        delete_summary = duplicates.delete()
+                        self.stdout.write('Removed duplicate memberships for {0}. Summary: {1}'.format(person, delete_summary))
```

Comment:
This can likely be done automatically, once we develop some confidence in it through testing.

Reply:
We can actually probably remove this when we run the full reimport on a fresh database, because there should not be duplicates within a single docket.